check.go

```diff
@@ -398,6 +414,21 @@ func (s *store) Check(options *CheckOptions) (CheckReport, error) {
 						expectedCheckDirectory.headers(diffHeaders)
 					}
 					// Scan the directory tree under the mount point.
+					var idmap *idtools.IDMappings
+					canUseShifting := true // mirror the logic in canUseShifting()
+					if !s.graphDriver.SupportsShifting() {
+						canUseShifting = false
+					}
+					if len(layer.UIDMap) != 0 && !idtools.IsContiguous(layer.UIDMap) {
+						canUseShifting = false
+					}
+					if len(layer.GIDMap) != 0 && !idtools.IsContiguous(layer.GIDMap) {
+						canUseShifting = false
+					}
```

Comment:
Is there a reason not to actually call `canUseShifting` as is?

If the code can’t be shared, maybe also a comment in `canUseShifting` to also update this location might be warranted.